### DISCUSSION TITLE
plot the stress-strain curve

### DISCUSSION THREAD

**Discussion by bright-men** [neperfepx/FEPX#116] [Q&A] [answered]
There was a problem when I finished running fepx according to the tutorials and output the stress-strain curve：
$ gnuplot plot_stress_strain.gp
"plot_stress_strain.gp" line 12: warning: Cannot find or open file "post.forces.z1"
"plot_stress_strain.gp" line 12: No data in plot

Besides, there is no file "post.forces.z1" when I finished simulation.

FEPX: Version 2.0.0
NEPER: Version 4.8.2

**Comment by tmivaniski** [comment]
Hello, I guess that you should change the file name. The results  in the simulation file is z0, z1... for example..  replace "post.forces.z1"  to "z1"

**Reply by mkasemer** [reply, marked as answer]
There are some known issues with regards to the tutorials and examples in the current FEPX version that we are working to fix.

In the meantime, you can print the equivalent stress and equivalent strain values as output, and use the `resmesh` option in `neper -S` to find the domain-averaged (i.e., macroscopic) values. These would be sufficient to plot the stress-strain curve.

**Reply by tmivaniski** [reply]
Could you write a simple example command here? 

Thank you very much

**Reply by mkasemer** [reply]
Of course. With a simulation complete (and having printed both `stress_eq` and `strain_eq`), try the following:

`neper -S /path/to/simulation.sim -resmesh stress_eq,strain_eq`

This should create a new directory with `mesh` level (domain-averaged) quantities: 

`/path/to/simulation.sim/results/mesh/`

This will contain your domain-averaged quantities, in a similar structure as previous (per-step files).

**Reply by tmivaniski** [reply]
Thank you, Sr.

Really appreciate it.

**Reply by bright-men** [reply]
Which folder should it be running in? I changed "post.forces.z1" to "z1". I put the file "plot_stress_strain.gp" in "../simple_simulation" or "/home/m/Documents/test/simple_simulation/simulation.sim/results/forces" and run it with "gnuplot plot_stress_strain.gp", but it didn't work.

**Reply by mkasemer** [reply]
As stated previously, there are known issues with some of the post processing scripts. If you would like to view the stress-strain curve, please follow the example I set above and use the values found in the `simulation.sim` folder to plot using your favorite method (Excel, Python, MATLAB...).

**Reply by bright-men** [reply]
Many thanks. Really appreciate it.
